### PR TITLE
Clean up command deduplication

### PR DESCRIPTION
**Pull request by rautenrieth-da**
This PR makes the following changes, each in its own commit:
- Change the submission service to always return `ALREADY_EXISTS` errors on duplicate submissions
- Remove unnecessary data about the result of previous submissions from the database
- Rename `TTL` to `deduplicationTime` or `deduplicateUntil`
- Change `domain.Commands` to store an absolute `deduplicateUntil` instead of a relative `deduplicationTime`

### CODE REVIEW

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/apiserver/services/ApiCommandService.scala [hunk 1]
```diff
@@ -194,7 +198,9 @@ object ApiCommandService {
       maxCommandsInFlight: Int,
       limitMaxCommandsInFlight: Boolean,
       historySize: Int,
-      retentionPeriod: FiniteDuration)
+      retentionPeriod: FiniteDuration,
+      // TODO(RC): this should be updated dynamically from the ledger configuration
```
Comment (gerolf-da) [inline ledger/sandbox/src/main/scala/com/digitalasset/platform/apiserver/services/ApiCommandService.scala]: hrhr, `RC` ;)

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/LedgerDao.scala [hunk 2]
```diff
@@ -144,20 +144,13 @@ trait LedgerReadDao extends ReportsHealth {
...
       submittedAt: Instant,
-      ttl: Instant): Future[Option[CommandDeduplicationEntry]]
-
-  /** Sets the result of a command, so that duplicate submissions can return the original result */
-  def updateCommandResult(
-      deduplicationKey: String,
-      submittedAt: Instant,
-      code: Int,
-      message: Option[String]): Future[Unit]
+      deduplicateUntil: Instant): Future[Option[CommandDeduplicationEntry]]
```
Comment (gerolf-da) [inline ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/LedgerDao.scala]: I think we can use the type `CommandDeduplicationResult` directly here. No need to add a layer of translation. What do you think?

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/apiserver/services/ApiCommandService.scala (hunk 1 above)
Comment (rautenrieth-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/apiserver/services/ApiCommandService.scala]: Force of habit 😄

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/LedgerDao.scala (hunk 2 above)
Comment (rautenrieth-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/LedgerDao.scala]: Initially, the ledger DAO tried to only use its own data types, but I see it already depends on `com.daml.ledger.participant.state.index.v2.PackageDetails`

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/apiserver/services/ApiCommandService.scala (hunk 1 above)
Comment (gerolf-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/apiserver/services/ApiCommandService.scala]: Apologies, I definitely didn't intend for you to change it :(
Just found it funny.

**Pull request merged**